servicebus/data-plane/azure-servicebus/src/main/java/com/microsoft/azure/servicebus/primitives/MessageWithDeliveryTag.java

```diff
@@ -5,21 +5,30 @@
 
 import org.apache.qpid.proton.message.Message;
 
+import java.util.Arrays;
+
 public class MessageWithDeliveryTag {
     private final Message message;
     private final byte[] deliveryTag;
 
-    public MessageWithDeliveryTag(Message message, byte[] deliveryTag)
-    {
+    public MessageWithDeliveryTag(Message message, byte[] deliveryTag) {
         this.message = message;
-        this.deliveryTag = deliveryTag;
```

Comment:
[SpotBugs-P2]
MessageWithDeliveryTag(Message, byte[]) may expose internal representation by storing an externally mutable object into MessageWithDeliveryTag.deliveryTag

This code stores a reference to an externally mutable object into the internal representation of the object.  If instances are accessed by untrusted code, and unchecked changes to the mutable object would compromise security or other important properties, you will need to do something different. Storing a copy of the object is better approach in many situations.